nginx/dev/adage-nginx.conf

```diff
@@ -0,0 +1,63 @@
+server {
+    # default server - drop requests without a Host header
+    listen 80 default_server;
+    server_name "";
+    return 444;
+}
+
+# configuration of the server
+server {
+    # the port your site will be served on
+    listen      80;
+    # the domain name it will serve for
+    server_name localhost;
+    charset     utf-8;
+    access_log on;
+
+    # max upload size
+    client_max_body_size 10M;   # adjust to taste
+
+    # /static is linked to static files required for use by the adage web application
+    location /static {
+        alias /home/static;
+    }
+
+    error_page   500 502 503 504  /50x.html;
+    location = /50x.html {
+        root   /usr/share/nginx/html;
+    }
+
+    # / directs to the index.html in the static folder for the
+    # adage web application
+    # for detail on the P3P header, see http://www.w3.org/P3P/
+    location / {
+        root /home/static;
+        proxy_set_header X-Forwarded-Host $server_name;
+        proxy_set_header X-Real-IP $remote_addr;
+        add_header P3P 'CP="ALL DSP COR PSAa PSDa OUR NOR ONL UNI COM NAV"';
+    }
+
+    # Set up redirecting for all the backend endpoints we want from our django
+    # server:
+
+    location /api {
+        proxy_pass http://adage-django:8000/api;
+        proxy_set_header X-Forwarded-Host $server_name;
+        proxy_set_header X-Real-IP $remote_addr;
+        add_header P3P 'CP="ALL DSP COR PSAa PSDa OUR NOR ONL UNI COM NAV"';
+    }
+
+    location /search {
+        proxy_pass http://adage-django:8000/search;
+        proxy_set_header X-Forwarded-Host $server_name;
+        proxy_set_header X-Real-IP $remote_addr;
+        add_header P3P 'CP="ALL DSP COR PSAa PSDa OUR NOR ONL UNI COM NAV"';
+    }
```

Comment:
Hmm. What is this? All of our searches should be handled via the `/api` above, right? Access to Elasticsearch would happen on its own separate port at `9200`.

Reply:
Yep, you're right @mhuyck - I had just copied all of the urls in `urls.py`, but the `/search` endpoint there was just a vestigial remnant from the haystack tutorial, probably, so I went ahead and removed it from there as well.